### PR TITLE
Seperated sandbox for taichi_core.so #501

### PR DESCRIPTION
**Pull request by archibate**
Related issue id = #501

### CODE REVIEW

**Comment by archibate** [comment]
Not done here, let's first complete opengl.

**Comment by yuanming-hu** [comment]
Cool! But please [fork from master instead of the OpenGL branch](https://taichi.readthedocs.io/en/latest/contributor_guide.html#making-good-pull-requests) to get rid of the unrelated OpenGL backend commits.

**Comment by archibate** [comment]
Sorry for making you confused :) Better now after rebased to master. Next event, do with Windows, seems have something different.

File: python/taichi/core/util.py [hunk 1]
```diff
@@ -159,6 +159,15 @@ def build():
 
   os.chdir(tmp_cwd)
 
+def prepare_sandbox(src):
+  assert os.path.exists(src)
+  from tempfile import mkdtemp
```
Comment (yuanming-hu) [inline python/taichi/core/util.py]: Wow, I didn't know we have a `tempfile` package that comes with Python! :-)

**Pull request merged**